src/main/java/com/yelp/nrtsearch/server/luceneserver/NRTReplicaNode.java

```diff
@@ -213,6 +217,9 @@ public CopyJob launchPreCopyFiles(
   @Override
   protected void finishNRTCopy(CopyJob job, long startNS) throws IOException {
     super.finishNRTCopy(job, startNS);
+    if (replicaDeleterManager != null) {
+      replicaDeleterManager.cleanUpReplicaFiles();
+    }
```

Comment:
Since we do not call ReplicaNode::commit, the initial commit files only really need to be decremented once. This could be done in the NRTReplicaNode::start method, which should be made synchronized. After calling super.start, read the current commit files out of the directory using the same logic as ReplicaNode::start. Decrementing the reference for these files should be all that needs to be done, as the current nrt point files will still be referenced.

Reply:
I'm writing test for it now. Quick question: I am trying to use [TestServer class](https://github.com/Yelp/nrtsearch/blob/master/src/test/java/com/yelp/nrtsearch/server/grpc/TestServer.java). But it seems it doesn't have the NRT functionality? I am trying to launch primary and replicas, send some docs to primary and commit refresh. Seems replica cannot see the updates?